docs/guides/migration.md

```diff
@@ -0,0 +1,341 @@
+
+In migrated resources, all child Managed Objects (MOs) are represented in the configuration as a map or a list of maps, rather than single attribute types (such as string, boolean, or integer). This approach ensures the configuration closely resembles the model, allowing for the addition of redefined attributes to a child without causing breaking changes.
+
+## Changed Child Objects in Payloads
+
+In non-migrated resources, child objects (including relationship objects) were managed via individual REST API requests, which could lead to a large number of REST API calls for a single resource containing many children. For migrated resources, a single GET/POST request that include all children objects is used to minimize the number of REST API calls made by each resource.
+
+## Changed Annotation Behavior
+
+The annotation attribute can be set for each MO that exposes the attribute, including children inside a resource. By default, the annotation is set to `orchestrator:terraform`, but this default can be overwritten at different levels (provider, resource, child attribute).
+
+```hcl
+provider "aci" {
+  username   = "admin"
+  password   = "password"
+  url        = "https://my-cisco-aci.com"
+  annotation = "provider-level-annotation-overwrite"
+}
+
+resource "aci_application_epg" "example" {
+  parent_dn  = aci_application_profile.example.id
+  name       = "test_name"
+  annotation = "resource-level-annotation-overwrite"
+  relation_to_bridge_domain = {
+    annotation         = "child-level-annotation-overwrite"
+    bridge_domain_name = aci_bridge_domain.example.name
+  }
+}
+```
+
+## Showing ID in Plan for Create
+
+For non-migrated resources, the ID of the resource appears as "known after apply". In the case of migrated resources, the ID is computed during the planning phase and included in the plan output to reduce the cascading attributes displaying a "known after apply".
+
+## Error for Existing MO on Create
+
+Migrated resources support the (`allow_existing_on_create`) provider-level option to check during the plan if an object already exists. By default, existing MOs will be automatically managed on apply, as (`allow_existing_on_create`) is set to `true` by default.  This option can be set to `false`, providing a mechanism to prevent the management of the same MO by separate configurations. The drawback of this mechanism is that it requires an additional API call per resource during the plan to verify that the object does not already exist.
+
+## Allowing Empty Input for Attributes
+
+Migrated resources in the Terraform ACI provider support null, unknown, and known states. Terraform supports three states for any value: null (missing), unknown ("known after apply"), and known. Previous SDKs did not support null and unknown states, which prevented differentiating between an empty value and a non-provided value. Because of this limitation, updating a string value to an empty string was not possible before the Plugin Framework.
```

Comment:
```suggestion
The attributes of migrated resources in the Terraform ACI provider now support null, unknown, and known states which can be used to reset values to empty or to enforce that children objects are not present. Terraform supports three states for any value: null (missing), unknown ("known after apply"), and known. Previous SDKs did not support null and unknown states, which prevented differentiating between an empty value and a non-provided value. Because of this limitation, updating a string value to an empty string, updating a map to an empty map or a list of maps to an empty list was not possible before the Plugin Framework but is now supported in the migrated resources.
```